src/player.cpp

```diff
@@ -3383,15 +3383,22 @@ void Player::doAttacking(uint32_t)
 		Item* tool = getWeapon();
 		const Weapon* weapon = g_weapons->getWeapon(tool);
 		if (weapon) {
+			uint32_t delay;
+
```

Comment:
if (weapon->interruptSwing() && !canDoAction()) {
				delay = getNextActionTime();
			} else {
				result = weapon->useWeapon(this, tool, attackedCreature);
				delay = getAttackSpeed();
				updateWeapon();
			}

			SchedulerTask* task = createSchedulerTask(delay, std::bind(&Game::checkCreatureAttack,
								  &g_game, getID()));
			setNextActionTask(task);